### PR TITLE
deletion of unused olio modules and functions; removal of deprecated functionality

### PR DESCRIPTION
**Pull request by andy-beer**
This change:

-  deletes olio modules that were unused by resqpy
- removes deprecated functionality

We can also remove the beta tag as part of this breaking change, in preparation for a major release tag (v2.0.0)

### CODE REVIEW

**Comment by andy-beer** [comment]
Note that this change is a **breaking change**, likely to require changes to application code.

File: resqpy/grid/_grid.py [hunk 1]
```diff
@@ -506,7 +492,10 @@ def off_handed(self):
         """Returns False if IJK and xyz have same handedness, True if they differ."""
 
         ijk_right_handed = self.extract_grid_is_right_handed()
-        assert rqet.find_tag_text(self.crs_root, 'ProjectedAxisOrder').lower() == 'easting northing'
+        if self.crs is None:
+            assert self.crs_uuid is not None
```
Comment (cflynn3) [inline resqpy/grid/_grid.py:496]: A general comment, but I much prefer to see a message after assertions like this to aid debugging if the situation occurs.

File: resqpy/olio/uuid.py [hunk 2]
```diff
@@ -194,3 +197,16 @@ def version_string(uuid_obj):
...
+    """Returns boolean indicating whether uuid_obj seems to be a uuid."""
+
+    if isinstance(uuid_obj, uuid.UUID):
+        return True
+    if not uuid_obj or not isinstance(uuid_obj, str):
+        return False
+    if uuid_obj[0] == '_':
+        return len(uuid_obj) == 37
+    return len(uuid_obj) == 36
+    # could also check for hyphens in correct places, and hexadecimals only
```
Comment (cflynn3) [inline resqpy/olio/uuid.py:212]: Agreed, I'm sure there must be some code somewhere that can validate a UUID fully.

**Review by cflynn3 — APPROVED** [review]
I've made a couple of general comments here, but I'm happy for this to be merged as is if you don't wish to act on them now.

File: resqpy/olio/uuid.py (hunk 2 above)
Comment (andy-beer) [reply resqpy/olio/uuid.py:212]: Yes, though I'm happy to let this go for now, as it is unlikely that something like-a-valid-UUID-but-not-quite will show up.

**Pull request merged**